### ISSUE TITLE
Issue with helper scripts when running Quick Start on macOS

### ISSUE THREAD

**Issue by msdolbey** [kubernetes-sigs/kubefed#1494]
**Issue #1**
**What happened**:
Two things happened when running the [Quick Start](https://github.com/kubernetes-sigs/kubefed#quickstart) on macOS: 
First, when running:

`make deploy.kind`

Somewhere down the line, I encountered this issue:
`go-bindata is not found. Use './scripts/download-binaries.sh' to download it.`

I run this script, and it seemed to download everything as one would expect.

However, if I run

`make deploy.kind`

again after that, I would get the same error.

**What you expected to happen**:
The dependency would download and be executable right away.

**How to reproduce it (as minimally and precisely as possible)**:
Run Quick start on Mac OS Big Sur - 11.6.3

**Anything else we need to know?**:
It seems that this fixes the issue:
https://github.com/go-bindata/go-bindata#installation

I added that bit to the script and it worked fine.

**Issue #2**
**What happened**:
Two things happened when running the [Quick Start](https://github.com/kubernetes-sigs/kubefed#quickstart) on macOS: 

Second thing that happened: 
On this command:

```
./scripts/fix-joined-kind-clusters.sh
```

This resulted in this errror:

```
./scripts/fix-joined-kind-clusters.sh
Error: No such object: kind-kind-control-plane
```

This looks like it's because it's calling this kind-kind-control-plane. Yet, the control-plane that is created when you build a kind cluster is named: 

```
kind-control-plane
```

This script doesn't seem to do what one would expect, because the control plane isn't named like that. 

It seems that a generic command of 

```
kind create cluster
```

will always result (at least in the version I'm using) with a kind cluster control plane having the name `kind-control-plane`

This could be hard-coded, I suppose in the `./scripts/fix-joined-kind-clusters.sh` script.

**What you expected to happen**:
This script will result in executing the patch

**How to reproduce it (as minimally and precisely as possible)**:
Run Quick start on Mac OS Big Sur - 11.6.3

**Anything else we need to know?**:
After changing the name of control plane in the script to `IP_ADDR="$(docker inspect -f "${INSPECT_PATH}" "kind-control-plane")"` - this all seemed to work. 

I did try this whole scenario again, and this step may be unnecessary for users of docker over a certain version? I'm not sure, but seems like this might be superfluous, but not sure.

**Environment**:
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:52:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-07-01T07:21:38Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```

- KubeFed version

```
kubefedctl version: version.Info{Version:"v0.7.0", GitCommit:"e13973ceaaba961a9abb6c7cb17a1d6a92c6eaf8", GitTreeState:"clean", BuildDate:"2021-04-01T15:31:47Z", GoVersion:"go1.15.3", Compiler:"gc", Platform:"darwin/amd64"}
```

- Scope of installation (namespaced or cluster)
n/a - just on getting started

- Others
Mac OS Big Sur - 11.6.3

/kind bug

**Comment by makkes** [comment]
@msdolbey, about:

> Somewhere down the line, I encountered this issue:
> go-bindata is not found. Use './scripts/download-binaries.sh' to download it.

Have you tried following the [instructions in the dev docs](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/development.md#binaries), especially the part about adding the `./bin` directory to `PATH`?

**Comment by msdolbey** [comment]
I believe I did - that didn't seem to help until I ran this bit first: 
`go get -u github.com/go-bindata/go-bindata/...`

that being said - there's always room for user error when I do a first run through of something. 

That being said - if this bit is used in the error:
`go-bindata is not found. Use './scripts/download-binaries.sh' to download it.`
Should it also instruct the user to then run:
`export PATH=$(pwd)/bin:${PATH}` ?

That might save a code/dependency bump change in the code?
